### ISSUE TITLE
Initialize model with pre-trained BART

### ISSUE THREAD

**Issue by chiyuzhang94** [justinlovelace/latent-diffusion-for-language#4]
Hi @justinlovelace ,

Thanks for your great work. 

I wonder if you try to use the weights of pre-trained BART to initialize your model. I didn't find these results in the paper while I can see you have a knob to keep pre-trained weights in the code. 

Best,
Chiyu

**Comment by justinlovelace** [comment]
Hello! Thank you for your interest!

We did not explore initializing the diffusion transformer with the weights from a pre-trained language model. Our codebase does not support that functionality. What knob you are referring to?

For the BART encoder and decoder, we freeze the pre-trained weights and do not update them during training. The only trainable parameters belong to the diffusion transformer.

**Comment by chiyuzhang94** [comment]
Hi @justinlovelace ,

I am so sorry. I sent the wrong question to you (this question is for another diffusion model). LoL
Sorry for the confusion.

My question for your paper is: have you used a pre-trained T5 instead of BART?

**Comment by justinlovelace** [comment]
No worries! [BART](https://arxiv.org/abs/1910.13461) is pre-trained to generate clean text given some corrupted text. Therefore, the pre-trained encoder's latent space is decoded by the pre-trained decoder into fluent text. We can therefore train our diffusion model directly in that latent space. 

[T5](https://arxiv.org/abs/1910.10683), on the other hand, is pre-trained to only generate the masked tokens given some corrupted text. You can see an example of T5's pre-training objective in the Huggingface documentation [here](https://huggingface.co/docs/transformers/model_doc/t5). Therefore, we wouldn't be able to repurpose T5's pre-training task directly. The encoder latent space is not actually decoded into fluent language by the pre-trained decoder. As a result, we focused on utilizing BART in our work.

**Comment by chiyuzhang94** [comment]
Thanks for the comprehensive answer! That makes sense.
